synapseclient/client.py

```diff
@@ -1354,6 +1355,115 @@ def uploadFileHandle(self, path, parent, synapseStore=True, mimetype=None, md5=N
         """
         return upload_file_handle(self, parent, path, synapseStore, md5, file_size, mimetype)
 
+    ############################################################
+    #                  Download List                           #
+    ############################################################
+    def clear_download_list(self):
+        """Clear all files from download list"""
+        self.restDELETE("/download/list")
+
+    def remove_from_download_list(self, list_of_files: typing.List[typing.Dict]) -> int:
+        """Remove a batch of files from download list
+
+        :param: array of files in the format of a mapping
+                {fileEntityId: synid, versionNumber: version}
+
+        :returns: Number of files removed from download list
+        """
+        request_body = {"batchToRemove": list_of_files}
+        num_files_removed = self.restPOST(
+            "/download/list/remove",
+            body=json.dumps(request_body)
+        )
+        return num_files_removed
+
+    def _generate_manifest_from_download_list(
+        self, quoteCharacter: str = '"',
+        escapeCharacter: str = "\\",
+        lineEnd: str = os.linesep,
+        separator: str = ",",
+        header: bool = True
+    ):
+        """Creates a download list manifest generation request
+
+        :param quoteCharacter:  The character to be used for quoted elements in the resulting file.
+                                Defaults to '"'.
+        :param escapeCharacter: The escape character to be used for escaping a separator or quote in the resulting
+                                file. Defaults to "\".
+        :param lineEnd:         The line feed terminator to be used for the resulting file. Defaults to os.linesep.
+        :param separator:       The delimiter to be used for separating entries in the resulting file. Defaults to ",".
+        :param header:          Is the first line a header? Defaults to True.
+
+        :returns: Filehandle of download list manifest
+        """
+        request_body = {
+            "concreteType": "org.sagebionetworks.repo.model.download.DownloadListManifestRequest",
+            "csvTableDescriptor": {
+                "separator": separator,
+                "quoteCharacter": quoteCharacter,
+                "escapeCharacter": escapeCharacter,
+                "lineEnd": lineEnd,
+                "isFirstLineHeader": header
+            }
+        }
+        return self._waitForAsync(uri="/download/list/manifest/async", request=request_body)
```

Comment:
Same as above.  Catching errors before returning could be a good implementation here.  Network requests are unpredictable (even when the API is made by us as well).

Reply:
There is error handling done in the client.  the respective Synapse errors will be returned